src/Tvheadend.cpp

```diff
@@ -1640,3 +1661,168 @@ uint32_t CTvheadend::GetNextUnnumberedChannelNumber()
   static uint32_t number = UNNUMBERED_CHANNEL;
   return number++;
 }
+
+void CTvheadend::TuneOnOldest( uint32_t channelId )
+{
+  int oldest = -1;
+
+  for (int i = 0; i < m_dmx.size(); i++)
+  {
+    if (m_dmx[i]->GetChannelId() == channelId)
+      return m_dmx[i]->Weight(PRETUNING_WEIGHT);
```

Comment:
Actually, this is more a syntactic sugar thing, as Weight() returns void as well. But I've splitted that up to follow with a _return;_ on that statement.